### PR TITLE
Listening to non-react updates to the preferences and update the react component

### PR DESCRIPTION
**Pull request by pooyaj**
Issue: Looks like users can use `savePreferences` to update the preferences in the cookie, but that does not update the internal state of our consent manager react component. 

Solution: Re-hydrate the state ( by calling reset ) when the dialog opens. 

Ref: https://segment.atlassian.net/browse/LIBWEB-482

### CODE REVIEW

**Review by juliofarah — APPROVED** [review]
It looks like you have a broken test there, but LGTM!

**Review by williamgrosset — APPROVED** [review]
LGTM once test is fixed!

**Review by nettofarah — APPROVED** [review]

**Pull request merged**
